app/javascript/react/utils/urlShortenedLink.ts

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+const useShortenedLink = (url: string, accessToken: string) => {
+  const [shortenedLink, setShortenedLink] = useState<string>(url);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    const shortenLink = async () => {
```

Comment:
Just to stay on the safe side we can add validation to check if url and access token are provided before taking action of shortening the link. WDYT?